app/bundles/AssetBundle/Model/AssetModel.php

```diff
@@ -124,6 +124,12 @@ public function trackDownload($asset, $request = null, $code = '200', $systemEnt
             $request = $this->requestStack->getCurrentRequest();
         }
 
+        if (null == $request) {
```

Comment:
As this is an ancient code without types, we should check that we have the right object rather than "not null"
```suggestion
        if ($request !instanceof RequestInterface) {
```
Same could be done on line 123